dependency.lic

```diff
@@ -772,257 +773,467 @@ class SlackbotManager
+      end
+    reload_profiles(include_filenames)
+    # Merge base and character settings
+    settings = [
+      'base.yaml',
+      'base-empty.yaml',
+      include_filenames,
+      character_filenames
+    ].flatten.reduce({}) do |result, filename|
+      result.merge(cache_get_by_filename(filename).data || {})
+    end
+    # Transform settings for quality of life improvements
+    transform_settings(settings)
   end
 
-  def pending?
-    @load
+  # Returns the config in a 'scripts/data/base-{type}.yaml' file.
+  # For example, type could be 'spells' or 'town' to return base-spells.yaml or base-town.yaml.
+  def get_data(type)
+    filename = to_base_filename(type)
+    # Ensure latest data is loaded into cache
+    reload_data([filename])
+    # Transform data for quality of life improvements
+    transform_data(cache_get_by_filename(filename).data)
   end
 
-  def request(options, data_load = false)
-    @data_load = data_load
-    @load = true
-    @extras = options
-    @settings = {}
+  # Designed to be called by dependency's run loop.
+  # Purpose is to check for file changes and reload them into cache.
+  def run_queue
+    reload_profiles(character_suffixes_to_filenames(['setup']))
+    reload_data
   end
 
-  def loaded?
-    !@load
+  private
+
+  SCRIPTS_DATA_PATH = File.join('.', 'scripts', 'data')
+  SCRIPTS_PROFILES_PATH = File.join('.', 'scripts', 'profiles')
+
+  # Reloads the base profiles plus any that match the filenames given.
+  def reload_profiles(filenames = [])
+    load_files(get_profiles_glob_patterns(filenames))
   end
 
-  def run_queue
-    load_base_settings if @load
-    load_data_settings if @load
-    load_settings if @load && !@data_load
-    load_data if @load && @data_load
-    save_settings if @save
+  # Reloads the base data plus any that match the filenames given.
+  def reload_data(filenames = [])
+    load_files(get_data_glob_patterns(filenames))
   end
 
-  def save_character_settings(contents)
-    @contents = contents
-    @save = true
+  # Gets glob patterns for the profiles directory.
+  # Includes base profiles plus any specified in the filenames argument.
+  def get_profiles_glob_patterns(filenames = [])
+    get_glob_patterns(SCRIPTS_PROFILES_PATH, filenames)
   end
 
-  def save_settings
-    profile = "./scripts/profiles/#{checkname}-setup.yaml"
-    File.open(profile, 'wb') { |file| file.puts(@contents.to_yaml) }
-    @save = false
+  # Gets glob patterns for the data directory.
+  # Includes base data plus any specified in the filenames argument.
+  def get_data_glob_patterns(filenames = [])
+    get_glob_patterns(SCRIPTS_DATA_PATH, filenames)
   end
 
-  attr_reader :settings
+  # Generates an array of file glob patterns, one per filename in the given basepath.
+  # `base*.yaml` is automatically and always included in the glob pattern.
+  def get_glob_patterns(basepath = '.', filenames = [])
+    filenames = ["base*.yaml", "include*.yaml", filenames].flatten.compact.uniq
```

Comment:
What is `include*.yaml`?

Reply:
This was news to me, but we have the ability to load include files: https://discordapp.com/channels/745675889622384681/745675890242879671/935919107944095755

Reply:
Oh wow, I totally forgot about that. Could you add a comment explaining it since it's 1) not in the repo; 2) not used by a bunch of people; and 3) I know I'm going to forget again. Thanks

Reply:
Will do

Reply:
Added comments about base and include yamls